File: Selenium/Project/Activity5.java
package HRM;


    import io.github.bonigarcia.wdm.WebDriverManager;
    import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

public class Activity5 {

        WebDriver driver;

        @BeforeTest
        public void setUp() {
            WebDriverManager.chromedriver().setup();
            //Create a new instance of the Firefox driver
            driver = new ChromeDriver();
            // Open the URL
            driver.get("http://alchemy.hguy.co/orangehrm");
        }

        @Test
        public void editUserInfo() {
            // Login with credentials
            driver.findElement(By.id("txtUsername")).sendKeys("orange");
            driver.findElement(By.id("txtPassword")).sendKeys("orangepassword123");
            driver.findElement(By.id("btnLogin")).click();

            // Click on "My Info" menu item
            driver.findElement(By.linkText("My Info")).click();

            // Click the Edit button
            driver.findElement(By.id("btnSave")).click();

            // Fill in the Name, Gender, Nationality, and DOB fields
            driver.findElement(By.id("personal_txtEmpFirstName")).clear();
            driver.findElement(By.id("personal_txtEmpFirstName")).sendKeys("John");

            driver.findElement(By.cssSelector("input[name='personal[txtEmpLastName]']")).clear();
            driver.findElement(By.cssSelector("input[name='personal[txtEmpLastName]']")).sendKeys("Doe");

            driver.findElement(By.id("personal_optGender_1")).click();

            driver.findElement(By.id("personal_cmbNation")).sendKeys("United States");

            driver.findElement(By.id("personal_DOB")).sendKeys("1990-01-01");

            // Click Save
            driver.findElement(By.id("btnSave")).click();
        }

        @AfterTest
        public void tearDown() {
            driver.quit();
        }
    }

File: Selenium/Project/Activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.chrome.service import Service as ChromeService
from webdriver_manager.chrome import ChromeDriverManager
driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))
# Navigate to the HRM system login page
driver.get("http://alchemy.hguy.co/orangehrm")

# Find the username and password input fields and login button
username_field = driver.find_element(By.ID, "txtUsername")
password_field = driver.find_element(By.ID, "txtPassword")
login_button = driver.find_element(By.ID, "btnLogin")

# Enter the login credentials
username_field.send_keys("orange")
password_field.send_keys("orangepassword123")

# Click the login button
login_button.click()

# Wait for the My Info menu item to become clickable in the menu
wait = WebDriverWait(driver, 10)
my_info_menu = wait.until(EC.element_to_be_clickable((By.ID, "menu_pim_viewMyDetails")))
my_info_menu.click()

# Click the Edit button
edit_button = driver.find_element(By.ID, "btnSave")
edit_button.click()

# Fill in the Name, Gender, Nationality, and DOB fields
name_field = driver.find_element(By.ID, "personal_txtEmpFirstName")
gender_dropdown = driver.find_element(By.ID, "personal_optGender_2")
nationality_dropdown = driver.find_element(By.ID, "personal_cmbNation")
dob_field = driver.find_element(By.ID, "personal_DOB")

# Replace the following placeholders with the desired information
name_field.clear()
name_field.send_keys("Updated Name")
gender_dropdown.click()  # Select the appropriate option in the dropdown
nationality_dropdown.send_keys("Nationality")  # Select the appropriate option in the dropdown
dob_field.clear()
dob_field.send_keys("2000-01-01")  # Format: YYYY-MM-DD

# Click the Save button
save_button = driver.find_element(By.ID, "btnSave")
save_button.click()
print("Saved")
# Close the browser
driver.quit()
